src/services/__tests__/anchor-service.test.ts

```diff
@@ -378,6 +378,23 @@ describe('anchor service', () => {
     expect(anchors.length).toEqual(numRequests)
   })
 
+  test('Retry persisting anchors on serializable erorr', async () => {
+    const requests = await fake.multipleRequests(4)
+
+    const [candidates] = await anchorService._findCandidates(requests, 0)
+    const merkleTree = await anchorService._buildMerkleTree(candidates)
+    const ipfsProofCid = await ipfsService.storeRecord({})
+    const anchors = await anchorService._createAnchorCommits(ipfsProofCid, merkleTree)
+
+    await Promise.all([
```

Comment:
I guess we're relying on a race condition here to *sometimes* trigger the serialization error we are catching and retrying on?  I would prefer if we inject the failure explicitly so it's deterministic, but also don't feel super strongly.  Can we at least add a comment explaining what we're doing here and why?

Reply:
made the test more deterministic and added comments